src/components/AccordionSection/AccordionSection.tsx

```diff
@@ -29,34 +29,29 @@ const AccordionSection = ({
   const sectionId = useRef(nanoid());
   const key = sectionKey || sectionId.current;
   const isExpanded = expanded === key;
-  const Title = titleElement || null;
-  const hasTitleElement = !!Title;
+  const Title = titleElement || "div";
 
   return (
     <li className="p-accordion__group">
-      <button
-        aria-controls={`#${sectionId.current}`}
-        aria-expanded={isExpanded ? "true" : "false"}
-        className={
-          hasTitleElement ? "p-accordion__tab--with-title" : "p-accordion__tab"
-        }
-        onClick={() => {
-          if (isExpanded) {
-            setExpanded(null, null);
-          } else {
-            setExpanded(key, title);
-          }
-          onTitleClick && onTitleClick(!isExpanded, key);
-        }}
-        role="tab"
-        type="button"
-      >
-        {hasTitleElement ? (
-          <Title className="p-accordion__title">{title}</Title>
-        ) : (
-          title
-        )}
-      </button>
+      <Title role="heading" className="p-accordion__heading">
```

Comment:
We should include the [`aria-level` attribute](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/examples/patterns/accordion/default.html#L10) here too, otherwise the heading has no semantic place in the document heirarchy when using a `div` rather than a heading element. 

I see this component currently accepts an optional `titleElement` prop, maybe we need to require a `headingLevel` prop too? I'm not sure - we wouldn't need it (or `role="heading"`, thinking about it) if `titleElement`  _was_ passed.

Reply:
I think the title element (assuming it is an `h*` doesn't need aria-level, probably even shouldn't have it). So it's only needed for divs.

I guess we can add a prop with some default value.